packages/plugin-npm/sources/npmHttpUtils.ts

```diff
@@ -259,7 +260,32 @@ async function whoami(registry: string, headers: {[key: string]: string} | undef
   }
 }
 
-async function askForOtp() {
+async function askForOtp(error: any, {configuration}: {configuration: Configuration}) {
+  const notice = error.originalError?.response.headers[`npm-notice`];
+
+  if (notice) {
+    await StreamReport.start({
+      configuration,
+      stdout: process.stdout,
+      includeFooter: false,
+    }, async report => {
+      report.reportInfo(MessageName.UNNAMED, notice.replace(/(https?:\/\/\S+)/g, formatUtils.pretty(configuration, `$1`, formatUtils.Type.URL)));
+    });
+
+    const autoOpen = notice.match(/open (https?:\/\/\S+)/i);
+    if (autoOpen) {
+      try {
+        await execUtils.execvp(`open`, [autoOpen[1]], {cwd: ppath.cwd()});
+      } catch {
+        try {
+          await execUtils.execvp(`xdg-open`, [autoOpen[1]], {cwd: ppath.cwd()});
+        } catch {}
+      }
+    }
```

Comment:
Poor Windows users, we should just let https://yarnpkg.com/package/open handle this so we don't have to.

Reply:
Doesn't `open` spawn a bundled shell file, meaning that we can't bundle it?

Reply:
It does but it will check if it can use it, if it can't it will use the system one.
https://github.com/sindresorhus/open/blob/05ba9e150cc1a2629e518a9cc19b586c6ca3f269/index.js#L173-L182

Reply:
I think the better solution would be to not auto open the link, this seems like a CVE waiting to happen.

Reply:
From a UX perspective I think the auto-open would still be very beneficial. What if we checked the hostname (or even the full url) to check it matches a known pattern? Various other tools have this kind of "open if possible" logic (AWS iirc?), so it should be possible to achieve it safely 🤔

Reply:
I've added a prompt; pressing `<enter>` will try to open the url, but people can use `n` (or ctrl-c).

Reply:
I still don't think we should if we can't do it consistently.

Reply:
Sorry, I saw your comment after merging - by consistently, do you refer to the Windows support? I can disable this altogether for win32, if you think that's better (I'm probably a little biased, but I'd like to avoid adding dependencies from this maintainer if I can avoid it ...).

Reply:
Yeah, referring to Windows support.